subset_vds.py

```diff
@@ -0,0 +1,373 @@
+    subset_vds = filter_intervals(input_vds, parsed_locus)
+    if subset_vds.variant_data.count_rows() == 0:
+        raise ValueError(
+            f"No rows remain after applying the following locus filters: {parsed_locus}"
+        )
+    return subset_vds
+
+
+def write_outputs(
+    output_formats: list[str],
+    subset_vds: VariantDataset | None,
+    subset_sample_list: list[str] | None,
+    infile_name: str,
+    is_test: bool,
+) -> None:
+    """Writes the vds out in the specified formats
+
+    Args:
+        output_formats (list[str]): list of output formats to write
+        subset_vds (VariantDataset | None): the resulting vds after subsetting by samples and / or intervals
+        subset_sample_list (list[str] | None): the samples left in the output data, when subsetting by samples
+        infile_name (str): the name of the input vds, to append '_subset' to
+        is_test (bool): controls whether to write to test or not
+
+    Raises:
+        FileExistsError: throws an error if any of the proposed output paths exist, as it will not overwrite them
+    """
+    if "vds" in output_formats:
+        if to_path(output_path(f"{infile_name}_subset")).exists():
+            raise FileExistsError(
+                f'The file {to_path(output_path("{infile_name}_subset"))} exists. Refusing to overwrite it.'
+            )
+        subset_vds.write(output_path(f"{infile_name}_subset", test=is_test))
+
+    subset_dense_mt: MatrixTable | Table | Any = to_dense_mt(subset_vds)
+    subset_dense_mt = split_multi_hts(subset_dense_mt)
+
+    for format in output_formats:
+        if format == "bed":
+            export_plink(
+                subset_dense_mt,
+                output_path(f"{infile_name}_subset", test=is_test),
+                call=subset_dense_mt.LGT,
+                ind_id=subset_dense_mt.s,
+            )
+        if format == "vcf":
+            if "gvcf_info" in subset_dense_mt.entry:
+                subset_dense_mt = subset_dense_mt.drop("gvcf_info")
+            export_vcf(
+                subset_dense_mt,
+                output_path(f"{infile_name}_subset.vcf.bgz", test=is_test),
+                tabix=True,
+            )
+
+    if subset_sample_list:
+        if to_path(output_path("subset_samples_file.txt")).exists():
+            raise FileExistsError(
+                f"The file {to_path(output_path('subset_samples_file.txt'))} exists. Refusing to overwrite it."
+            )
+        outdata = StringIO()
+        for single_sample in subset_sample_list:
+            outdata.write(f"{single_sample}\n")
+        with to_path(output_path("subset_samples_file.txt", test=is_test)).open(
+            "wt"
+        ) as outfile:
+            outfile.write(outdata.getvalue())
+            outdata.close()
+
+
+def main(
+    vds_path: str,
+    n_samples: int | None,
+    intervals: str | None,
+    output_formats: list[str],
+) -> None:
+    allowed_output_formats: list[str] = ["vcf", "bed", "vds"]
```

Comment:
you can put this validation in the argparse interface
https://docs.python.org/3/library/argparse.html#choices

Reply:
wait, no you can't... not if you tolerate multiple. Curses.

Reply:
ooooh, or can you. You could have a list of choices, set `nargs='+'` on the argument, and take multiple individual arguments. that would remove the need for 
```output_formats=args.output_formats.lower().split(",")```

Reply:
Ah, that was what I was wanting to do, I assume then it's just multiple space separated arguments to --output-formats?